vizro-core/docs/pages/user-guides/dynamic-data.md

```diff
@@ -0,0 +1,103 @@
+            title="My first page",
+            components=[
+                vm.Graph(figure=px.scatter("iris", x="sepal_length", y="petal_width", color="species")), # (2)!
+            ],
+            controls=[vm.Filter(column="species")],
+        )
+
+        dashboard = vm.Dashboard(pages=[page])
+
+        Vizro().build(dashboard).run()
+        ```
+
+        1. To use `load_iris_data` as dynamic data it must first be added to the Data Manager. You should **not** actually call the function as `load_iris_data()`; doing so would result in static data that cannot be reloaded.
+        2. Dynamic data is referenced by the name of the data source `"iris"`.
+
+    === "Result"
+        [![DataBasic]][DataBasic]
+
+    [DataBasic]: ../../assets/user_guides/data/data_pandas_dataframe.png
+
+Since dynamic data sources must always be added to the Data Manager and referenced by name, they may be used in YAML configuration [exactly the same way as for static data sources](static-data.md#reference-by-name).
+
+## Configure cache
+
+By default, every time the dashboard is refreshed a dynamic data function will be executed again. This means that your dashboard will always show the very latest data. In fact, if there are multiple graphs on the same page using the same dynamic data source then the loading function will be executed _multiple_ times, once for each graph on the page. Hence, if loading your data is a slow operation, your dashboard performance may suffer.
+
+The Vizro Data Manager has a server-side caching mechanism to help solve this. Vizro's cache uses [Flask-Caching](https://flask-caching.readthedocs.io/en/latest/), which supports a number of possible cache backends and [configuration options](https://flask-caching.readthedocs.io/en/latest/#configuring-flask-caching). By default, the cache is turned off. 
+
+In a development environment the easiest way to enable caching is to use a [simple memory cache](https://cachelib.readthedocs.io/en/stable/simple/) with the default configuration options:
+
+```py title="Simple cache with default timeout of 5 minutes"
+from flask_caching import Cache
+
+data_manager.cache = Cache(config={"CACHE_TYPE": "SimpleCache"})
+data_manager["iris"] = load_iris_data
+```
+
+By default, dynamic data is cached in the Data Manager for 5 minutes. A refresh of the dashboard within this time interval will fetch the pandas DataFrame from the cache and not re-run the data loading function. Once the cache timeout period has elapsed, the next refresh of the dashboard will re-execute the dynamic data loading function. The resulting pandas DataFrame will again be put into the cache and not expire until another 5 minutes has elapsed.
+
+If you would like to alter some options, such as the default cache timeout, then you can specify a different cache configuration:
+
+```py title="Simple cache with timeout set to 10 minutes"
+data_manager.cache = Cache(config={"CACHE_TYPE": "SimpleCache", "CACHE_DEFAULT_TIMEOUT": 600})
```

Comment:
Is it possible to set the cache to just refresh when the source has been modified, e.g. when it detects a file modification since the last load?

I am thinking here of the model interactions case - when the kedro run has finish, then the charts should be reloaded

Reply:
Hmm, it's probably possible to update cache automatically when a file changes but not very easy.

What will be easy though, and coming in the near future in a different PR, is to have an action that forcibly updates data without waiting for it to expire. So you could have something that does:
1. action to trigger kedro run
2. action to refresh data